File: backend/models/FalseNegativeReport.js
module.exports = (sequelize, DataTypes) => {
  const FalseNegativeReport = sequelize.define('FalseNegativeReport', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    userId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: { model: 'Users', key: 'id' },
    },
    soundType: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    description: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
    occurredAt: {
      type: DataTypes.DATE,
      allowNull: false,
    },
    deviceInfo: {
      type: DataTypes.STRING,
      allowNull: true,
    },
    status: {
      type: DataTypes.ENUM('open', 'reviewed', 'resolved'),
      defaultValue: 'open',
      allowNull: false,
    },
    reviewedBy: {
      type: DataTypes.INTEGER,
      allowNull: true,
      references: { model: 'Users', key: 'id' },
    },
    adminNotes: {
      type: DataTypes.TEXT,
      allowNull: true,
    },
  }, {
    timestamps: true,
    tableName: 'FalseNegativeReports',
  });

  FalseNegativeReport.associate = (db) => {
    FalseNegativeReport.belongsTo(db.User, { foreignKey: 'userId', as: 'user' });
    FalseNegativeReport.belongsTo(db.User, { foreignKey: 'reviewedBy', as: 'reviewer' });
  };

  return FalseNegativeReport;
};

File: app/lib/models/false_negative_report.dart
class FalseNegativeReport {
  final int? id;
  final String? soundType;
  final String? description;
  final DateTime occurredAt;
  final String? deviceInfo;

  FalseNegativeReport({
    this.id,
    this.soundType,
    this.description,
    required this.occurredAt,
    this.deviceInfo,
  });

  Map<String, dynamic> toJson() => {
    if (soundType != null) 'soundType': soundType,
    if (description != null) 'description': description,
    'occurredAt': occurredAt.toIso8601String(),
    if (deviceInfo != null) 'deviceInfo': deviceInfo,
  };
}
